### PR TITLE
 Remove workaround for dependency issues and missing patterns

### PR DESCRIPTION
**Pull request by rwx788**
Now issues with dependencies should not appear anymore and hence we
should not soft-fail them. Point here is that original bug was created
for leanOS. If we test SLES, we should create relevant bug. To resolve
original issue, we add modules which are expected to be in installation.

We still continue adding modules manually instead of using SCC
functionality, recently we have introduced another variable to add
specific modules only. We also move it to addon_products_sle test
module where behavior has workaround applied.

[Verification run](http://gershwin.arch.suse.de/tests/1376#settings).
See [poo#23938](https://progress.opensuse.org/issues/23938).

### CODE REVIEW

**Comment by rwx788** [comment]
In latest build we again have dependency issues: https://openqa.suse.de/tests/1153562#step/installation_overview_before/5

**Comment by coolo** [comment]
and it's good to report them asap because the build is garbage :)

**Comment by rwx788** [comment]
@coolo it's a shame, but we don't see anything else so start forgetting that it can be different =)

File: tests/installation/addon_products_sle.pm [hunk 1]
```diff
@@ -20,6 +20,12 @@ use qam 'advance_installer_window';
 sub run {
     my ($self) = @_;
 
+    # Softfail not to forget remove workaround
+    # TODO: Remove once fixed for SLE15
+    if (get_var('ALL_MODULES') || get_var('WORKAROUND_MODULES')) {
+        record_soft_failure('bsc#1054974');
```
Comment (okurz) [inline tests/installation/addon_products_sle.pm]: the bug is already VERIFIED FIXED, we should get rid of this as well

**Comment by sysrich** [comment]
Needs a rebase

**Comment by okurz** [comment]
I assume that PR is obsoleted by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3599

**Pull request closed without merging**